plugin/pkg/admission/initialresources/admission.go

```diff
@@ -138,29 +143,47 @@ func (ir initialResources) estimateAndFillResourcesIfNotSet(pod *api.Pod) {
 	}
 }
 
-func (ir initialResources) getEstimation(kind api.ResourceName, c *api.Container) (*resource.Quantity, error) {
+func (ir initialResources) getEstimation(kind api.ResourceName, c *api.Container, ns string) (*resource.Quantity, error) {
 	end := time.Now()
 	start := end.Add(-week)
 	var usage, samples int64
 	var err error
 
-	// Historical data from last 7 days for the same image:tag.
-	if usage, samples, err = ir.source.GetUsagePercentile(kind, *percentile, c.Image, true, start, end); err != nil {
+	// Historical data from last 7 days for the same image:tag within the same namespace.
+	if usage, samples, err = ir.source.GetUsagePercentile(kind, ir.percentile, c.Image, ns, true, start, end); err != nil {
 		return nil, err
 	}
 	if samples < samplesThreshold {
-		// Historical data from last 30 days for the same image:tag.
+		// Historical data from last 30 days for the same image:tag within the same namespace.
 		start := end.Add(-month)
-		if usage, samples, err = ir.source.GetUsagePercentile(kind, *percentile, c.Image, true, start, end); err != nil {
+		if usage, samples, err = ir.source.GetUsagePercentile(kind, ir.percentile, c.Image, ns, true, start, end); err != nil {
 			return nil, err
 		}
 	}
-	if samples < samplesThreshold {
-		// Historical data from last 30 days for the same image.
-		start := end.Add(-month)
-		image := strings.Split(c.Image, ":")[0]
-		if usage, samples, err = ir.source.GetUsagePercentile(kind, *percentile, image, false, start, end); err != nil {
-			return nil, err
+
+	// If we are allowed to estimate also based on data from other namespaces.
+	if !ir.nsAware {
+		if samples < samplesThreshold {
+			// Historical data from last 7 days for the same image:tag within all namespaces.
+			start := end.Add(-week)
+			if usage, samples, err = ir.source.GetUsagePercentile(kind, ir.percentile, c.Image, "", true, start, end); err != nil {
+				return nil, err
+			}
+		}
+		if samples < samplesThreshold {
+			// Historical data from last 30 days for the same image:tag within all namespaces.
+			start := end.Add(-month)
+			if usage, samples, err = ir.source.GetUsagePercentile(kind, ir.percentile, c.Image, "", true, start, end); err != nil {
+				return nil, err
+			}
+		}
+		if samples < samplesThreshold {
```

Comment:
there should also be same image (no tag) within the specific namespace, if ir.nsAware/ns.Only